apps/login/ui/UsernameForm.tsx

```diff
@@ -34,22 +49,77 @@ export default function UsernameForm() {
 
     setLoading(false);
     if (!res.ok) {
-      throw new Error("Failed to set user");
+      throw new Error("Failed to load authentication methods");
     }
     return res.json();
   }
 
-  function submitUsernameAndContinue(value: Inputs): Promise<boolean | void> {
-    return submitUsername(value).then(({ factors }) => {
-      return router.push(
-        `/password?` +
-          new URLSearchParams({ loginName: `${factors.user.loginName}` })
-      );
+  async function setLoginNameAndGetAuthMethods(values: Inputs) {
+    return submitLoginName(values).then((response) => {
+      if (response.authMethodTypes.length == 1) {
+        const method = response.authMethodTypes[0];
+        switch (method) {
+          case 1: //AuthenticationMethodType.AUTHENTICATION_METHOD_TYPE_PASSWORD:
```

Comment:
Does the incommented value not work?

Reply:
The commented code imports other serverside components, that result in an error as this `UsernameForm.tsx` is executed in the browser